lib/Sema/ConstraintSystem.cpp

```diff
@@ -3743,6 +3747,14 @@ static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
     llvm::SmallSetVector<Type, 4> arguments;
     for (const auto &solution : solutions) {
       auto type = solution.typeBindings.lookup(typeVar);
+      // Type variables gathered from a solution's type binding context may not
+      // exist in another given solution because each solution was formed after
+      // solver has explored different scopes with different type variables
+      // being created and bound while solving the system at each one of those
+      // scopes.
```

Comment:
Sounds good but I'd simplify it a bit to something like - Each solution could have different set of overload choices picked, which implies that some of the generic parameters, and type variables that represent them, would be unique to that solution.

Reply:
It makes sense, just adjusted :)